util/codec/codec.go

```diff
@@ -52,7 +53,18 @@ func encode(b []byte, vals []types.Datum, comparable bool) ([]byte, error) {
 			b = encodeBytes(b, val.GetBytes(), comparable)
 		case types.KindMysqlTime:
 			b = append(b, uintFlag)
-			v, err := val.GetMysqlTime().ToPackedUint()
+			t := val.GetMysqlTime()
+			// Encoding timestamp need to consider timezone.
+			// if it's not in UTC, transform to UTC first.
+			if t.Type == mysql.TypeTimestamp && t.TimeZone != time.UTC && !t.IsZero() {
```

Comment:
Why this [line](https://github.com/pingcap/tidb/blob/master/tablecodec/tablecodec.go#L192)  doesn't consider `t.TimeZone != time.UTC`?